stdlib/xml/etree/ElementTree.pyi

```diff
@@ -259,11 +260,33 @@ def fromstringlist(sequence: Sequence[str | bytes], parser: XMLParser | None = .
 _ElementFactory = Callable[[Any, Dict[Any, Any]], Element]
 
 class TreeBuilder:
-    def __init__(self, element_factory: _ElementFactory | None = ...) -> None: ...
+    if sys.version_info >= (3, 8):
+        # First arg of `comment_factory` has to take `str | None`
+        # because `self._comment_factory = Comment if comment_factory is None else comment_factory` in `TreeBuilder.__init__`
+        def __init__(
+            self,
+            element_factory: _ElementFactory | None = ...,
+            *,
+            comment_factory: Callable[[str | None], Element] | None = ...,
+            pi_factory: Callable[[str, str | None], Element] | None = ...,
+            insert_comments: bool = ...,
+            insert_pis: bool = ...,
+        ) -> None: ...
+        insert_comments: bool
+        insert_pis: bool
+    else:
+        def __init__(self, element_factory: _ElementFactory | None = ...) -> None: ...
     def close(self) -> Element: ...
     def data(self, __data: str | bytes) -> None: ...
     def start(self, __tag: str | bytes, __attrs: dict[str | bytes, str | bytes]) -> Element: ...
     def end(self, __tag: str | bytes) -> Element: ...
+    if sys.version_info >= (3, 8):
+        # `text` is pos-only in the C implementation
+        # First arg of `comment_factory` has to take `str | None`
+        # because `self._comment_factory = Comment if comment_factory is None else comment_factory` in `TreeBuilder.__init__`
```

Comment:
I don't think we need this comment, because the implementation quite clearly passes it to `self._comment_factory` as is.